src/auth/strategies/jwt.strategy.ts

```diff
@@ -0,0 +1,23 @@
+import { Injectable } from '@nestjs/common'
+import { ConfigService } from '@nestjs/config'
+import { PassportStrategy } from '@nestjs/passport'
+import { ExtractJwt, Strategy } from 'passport-jwt'
+
+import { throwMissingEnvVar } from '../../common/utils/env.utils'
+
+@Injectable()
+export class JwtStrategy extends PassportStrategy(Strategy) {
+  constructor(configService: ConfigService) {
+    super({
+      jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
```

Comment:
Shouldn't we use `bind` here in order to prevent `this` loose?

Reply:
But we call `super` withing the `constructor`, does we really should to use `bind` here in addition? As I know, all injections are handled by NestJS and it looks like an overcontrol from our side

Reply:
There was nothing related to NestJS nor `super` in my question.
Methods are losing `this` during assignment to the variables or passing as argument. If `fromAuthHeaderAsBearerToken` is a method of `ExtractJwt`, it will lose `this` during assignment to the `jwtFromRequest` object property in the sample below:
```TS
{
  jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken,
}
```

But now I see that we're _calling_ `fromAuthHeaderAsBearerToken`, so please disregard my question.